component/loki/source/kubernetes/kubetail/tailer.go

```diff
@@ -167,9 +169,51 @@ func (t *tailer) tail(ctx context.Context, handler loki.EntryHandler) error {
 	if err != nil {
 		return err
 	}
+
+	// Create a new rolling average calculator to determine the average delta
+	// time between log entries.
+	//
+	// Here, we track the most recent 10,000 delta times to compute a fairly
+	// accurate average. If there are less than 100 deltas stored, the average
+	// time defaults to 1h.
+	//
+	// The computed average will never be less than the minimum of 2s.
+	calc := newRollingAverageCalculator(10000, 100, 2*time.Second, maxTailerLifetime)
```

Comment:
> thats a good chunk of memory for each target. 

It's 80KB per target, right? A time.Duration is a uint64 (8 bytes), meaning this is 80,000 bytes per target. 

I'm not sure I'm convinced yet that 80KB is a huge overhead for a single target, especially since once you have thousands of targets I'm not sure this would be the largest portion of the agent's memory usage. 

We can also tune this over time if the overhead does appear to be significant; I'm not sure if there was any particular reason @slim-bean chose these specific values. 

> If there was a good way to detect k8s version or something, it could reduce memory at the cost of complexity. Maybe not worth it?

I'd be happy to do this, but I'm not sure that the fix is released yet?